lib/ansible/plugins/action/__init__.py

```diff
@@ -346,8 +346,11 @@ def _execute_module(self, module_name=None, module_args=None, tmp=None, task_var
         # set no log in the module arguments, if required
         if self._play_context.no_log:
             module_args['_ansible_no_log'] = True
+            module_display_args = "(no_log enabled, args censored)"
+        else:
+            module_display_args = module_args
 
-        self._display.vvv("Module: %s, args: %s" % (module_name, module_args))
+        self._display.vvv("Module: %s, args: %s" % (module_name, module_display_args))
```

Comment:
That's a larger problem for a lot of reasons- one of the things I was going to suggest was implementing some declarative module metadata in terms of the doc structure (or something like it) so that the runner-side can do arg validation and modify its behavior to deal with things like this. In the face of non-Python modules, it's pretty much impossible without having some sort of non-executable arg metadata.